src/test/java/SortingTest.java

```diff
@@ -0,0 +1,86 @@
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Created by sam on 6/17/16.
+ */
+public class SortingTest {
+
+    // to create an heapArray of 10 random ints between 0 and 100 for use in tests
+    private List<Integer>  randomArray() {
+        List<Integer> array = new ArrayList<Integer>();
+
+        for(int i = 0; i < 10; i++) {
+            array.add((int) (Math.random() * 100));
+        }
+
+        return array;
+    }
+    
+    
+    private List<Integer> sort(List<Integer> list) {
+//        return Sorting.bubbleSort(list);
```

Comment:
here's your challenge: none of these should be commented out. either we duplicate the test cases for each sorting algorithm (testBubbleSort1, testBubbleSort2, ...) or we find a creative way to invoke those methods. i will leave that up to you. normally i hate reflection, but in this case we could cut down on a substantial amount of duplicated test cases.